File: Python/2194.py
# 2194. Cells in a Range on an Excel Sheet
# https://leetcode.com/problems/cells-in-a-range-on-an-excel-sheet/

class Solution:
  def cellsInRange(self, s: str) -> List[str]:
    out = []

    row_beg = ord(s[0])
    col_beg = int(s[1])
    row_end = ord(s[3])
    col_end = int(s[4])

    for i in range(row_beg, row_end + 1):
      for j in range(col_beg, col_end + 1):
        out.append(chr(i) + str(j))

    return out

File: Rust/2194.rs
/**
 * 2194. Cells in a Range on an Excel Sheet
 * https://leetcode.com/problems/cells-in-a-range-on-an-excel-sheet/
 */

impl Solution {
  pub fn cells_in_range(s: String) -> Vec<String> {
    let s = s.chars().collect::<Vec<_>>();
    let mut out: Vec<String> = Vec::new();

    let row_beg: u8 = s[0] as u8;
    let col_beg: u8 = s[1].to_digit(10).unwrap() as u8;
    let row_end: u8 = s[3] as u8;
    let col_end: u8 = s[4].to_digit(10).unwrap() as u8;

    for i in row_beg..=row_end {
      for j in col_beg..=col_end {
        out.push(format!("{}{}", char::from(i), j));
      }
    }

    return out;
  }
}
